### ISSUE TITLE
CRS: Reorganize - All Files error

### ISSUE THREAD

**Issue by nicolassaleron** [CloudReadySoftware/crs-al-language-extension#110]
Hello,

Since version [1.1.0] - 2019-03-26, the action "CRS: Reorganize - All Files" returns the error "Cannot read property 'uri' of undefined".

**Comment by GreatScott000** [comment]
I receive the same error using CRS: Rename - All Files.  It happens whether the new setting `CRS.RenameWithGit` is false or true.

Also, with this setting TRUE I don't see any staged objects in source control (Azure DevOps using Git).

Does this setting apply to Rename - Single file too?

**Comment by nicolassaleron** [comment]
By the way, renaming one file at a time works fine.

**Comment by waldo1001** [comment]
Interesting.  I have renamed and reorganized all files in the baseapp (which means +5000 files), and it worked fine, both with Git and without git.  So I'm not able to reproduce.

Can you share a usecase with me?  Maybe a sample project?  It can depend on:
- how you setup your settings (workspace, multiple workspaces, user, ...)
- characters in your file
- ...

would be great, because else, I'm searching blind.. :(

**Comment by waldo1001** [comment]
@nicolassaleron & @GreatScott000 ,

hm, one thing you might try (judging from the error): open a file from the workspace, and THEN rename all.

**Comment by nicolassaleron** [comment]
I have attached a sample project (quite simple I think :) )

When you use "CRS: Reorganize - All Files", MyTable.al is not moved or renamed.

Note that if you delete the folder named "src", il will be correctly created (and the subfolder will be also correctly created).

I am running the last version of VSCode (1.32.3).

[TestProject.zip](https://github.com/CloudReadySoftware/crs-al-language-extension/files/3016788/TestProject.zip)

**Comment by GreatScott000** [comment]
I happen to have a Bash terminal session open and saw this when I try to rename a single file.

```
Todd.Scott@5CG70948WR MINGW64 /c/EvolveiMS/Repos/dev/Nav/DRON001/13.0 (Nav BC)/Extension/Legacy (master)
$ code c:\EvolveiMS\Repos\dev\Nav\DRON001\13.0 (Nav BC)\Extension\Legacy\PageExtension\Interaction Log Entries Ext 123.albash: syntax error near unexpected token `('
```
Note, there is a ( and ) in my folder path.  

The file does get renamed but not using the `git mv` command noted in the the release notes.  There is no staged file.

**Comment by waldo1001** [comment]
@GreatScott000 & @nicolassaleron ,

I'm still not able to reproduce, even not with the attached app.  but I cleaned up some errors in the background (similar).  May be you could test this?  You can download the new version here (not on the marketplace yet):
https://www.dropbox.com/s/auaso0zig9s2erd/crs-al-language-extension-1.1.1.vsix?dl=1

**Comment by nicolassaleron** [comment]
thanks, works fine for me now :)

By my mistake, I tried the new version with an empty al file in my project and get the error message "Cannot read property 'toLowerCase' of undefined".

**Comment by waldo1001** [comment]
> By my mistake, I tried the new version with an empty al file in my project and get the error message "Cannot read property 'toLowerCase' of undefined".

@nicolassaleron,
hm, again, not able to repro that one.  If you allow me, I will neglect it for now.  If it's really an issue, let's create a new issue on github for this ..

**Comment by fgilsc** [comment]
I'm experimenting this error for a few days ago
![image](https://user-images.githubusercontent.com/15670597/55151011-9ffcfe80-514d-11e9-953b-39b7f7ce50a6.png)

**Comment by GreatScott000** [comment]
Okay, the Bash error is gone but I still do not see the `git mv`

This is the Output from Git with I run CRS: Rename - Single File

> Looking for git in: C:\Program Files\Git\cmd\git.exe
> Using git 2.20.1.windows.1 from C:\Program Files\Git\cmd\git.exe
> > git rev-parse --show-toplevel
> > git config --get commit.template
> Open repository: c:\EvolveiMS\Repos\dev
> > git fetch
> > git status -z -u
> > git symbolic-ref --short HEAD
> > git rev-parse master
> > git rev-parse --symbolic-full-name master@{u}
> > git check-ignore -z --stdin
> > git rev-list --left-right master...refs/remotes/origin/master
> > git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> > git remote --verbose
> > git status -z -u
> > git symbolic-ref --short HEAD
> > git rev-parse master
> > git rev-parse --symbolic-full-name master@{u}
> > git rev-list --left-right master...refs/remotes/origin/master
> > git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> > git remote --verbose
> > git check-ignore -z --stdin
> > git show :Nav/DRON001/13.0 (Nav BC)/Extension/Legacy/Page/Attendee Interact Wiz. Subform 50088.al
> > git status -z -u
> > git check-ignore -z --stdin
> > git symbolic-ref --short HEAD
> > git rev-parse master
> > git rev-parse --symbolic-full-name master@{u}
> > git rev-list --left-right master...refs/remotes/origin/master
> > git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> > git remote --verbose
> > git show :Nav/DRON001/13.0 (Nav BC)/Extension/Legacy/Page/Attendee Interact Wiz. Subform 50500.al
> > git ls-files --stage -- Nav/DRON001/13.0 (Nav BC)/Extension/Legacy/Page/
> 
This is the crs-al-language output

> 28/03/2019, 10:35:52 - Rename file from Attendee Interact Wiz. Subform 50088.al to Attendee Interact Wiz. Subform 50500.al

And this is what I see in Source Control: GIT
![image](https://user-images.githubusercontent.com/4333956/55151267-c323b000-5145-11e9-95f4-ba20b8b3c5c8.png)

If I understood the release notes correct I should see something in Staged Changes, correct?

**Comment by waldo1001** [comment]
@GreatScott000 , I'm sure you did set the setting ""CRS.RenameWithGit": true".  Where did you do that?  What settings.json?

**Comment by waldo1001** [comment]
> I'm experimenting this error for a few days ago
> ![image](https://user-images.githubusercontent.com/15670597/55151011-9ffcfe80-514d-11e9-953b-39b7f7ce50a6.png)

Is it possible to share a bit more? what situation?  Settings?  the project?  A sample?

**Comment by GreatScott000** [comment]
@waldo1001 
It is in my User Settings.
![image](https://user-images.githubusercontent.com/4333956/55166191-916f1100-5166-11e9-8cff-050584f88321.png)
I tried with it turned on in both User and Workspace (only opening a folder), same results.

**Comment by GreatScott000** [comment]
@waldo1001 I just tried it with it only on in Workspace, same problem.

**Comment by waldo1001** [comment]
This is so weird :(.  I did notice I have to "Reload Window" after I have changed the setting to make it work, but still .. .
Stupid question: your repo is initialized for git, right? ;-)

**Comment by waldo1001** [comment]
I sensed a disturbance in the force when working with multiple workspaces.  

I suspect you have multiple workspaces - true?  can you test by opening just one workspace?

I need to fix this, of course, but just like to see if I'm on the right path ;-)

**Comment by waldo1001** [comment]
@GreatScott000 

OK, i **think** i solved the multiple workspaces part (by not checking it's a git repo or not).

I would greatly appreciate if you could test again!
https://www.dropbox.com/s/auaso0zig9s2erd/crs-al-language-extension-1.1.1.vsix?dl=1

**Comment by GreatScott000** [comment]
Yes, I have multiple folders in a workspace and as far as I know it is a GIT repository.  It shows SOURCE CONTROL: GIT in the window anyway!

I installed the new VSIX you supplied.  I first had to uninstall the AL Outline extension.  I then decided to uninstall the previous version of CRS-AL-Language too.

I manually changed my settings.json (in .vscode) `"CRS.RenameWithGit": true` rather than using the settings editor.  Renamed my file and....

It works!

I did this first in a single folder rather than opening my workspace.

I then updated my workspace settings file and IT WORKS!

Thanks @waldo1001 love this tool
